content/kubernetes-platform-services/artifact-storage/chartmuseum.md

```diff
@@ -0,0 +1,122 @@
+
+### Install with Master Helmfile
+
+Master Helmfile provide two releases of chartmuseum:
+* `charts` - Chartmuseum that serve charts
+* `charts-api` - Chartmuseum that provide api gateway to publish charts.
+
+These releases shares the same environment variables.
+`charts-api` gateway would be available on subdomain `api` for FQHN.
+In our example it would `api.charts.us-west-2.staging.example.com`.
+
+To install releases follow these instructions:
+1. Set the `CHARTMUSEUM_STORAGE_AMAZON_BUCKET` secret with chamber to copied from output value
+2. Set the `CHARTMUSEUM_STORAGE_AMAZON_REGION` secret with chamber
+3. Set the `CHARTMUSEUM_IAM_ROLE` secret with chamber to copied from output value
+4. Set the `CHARTMUSEUM_INGRESS` secret with chamber provided by [Nginx ingress]({{< relref "kubernetes-backing-services/ingress/nginx-ingress-controller.md" >}})
+5. Set the `CHARTMUSEUM_HOSTNAME` secret with chamber
+6. Run command `helmfile sync` to install `chartmuseum`.
+
+{{% dialog type="code-block" icon="fa fa-code" title="Install chartmuseum" %}}
+```
+chamber write kops CHARTMUSEUM_STORAGE_AMAZON_BUCKET example-staging-chart-repo
+chamber write kops CHARTMUSEUM_STORAGE_AMAZON_REGION us-west-2
+chamber write kops CHARTMUSEUM_IAM_ROLE example-staging-chart-repo
+chamber write kops CHARTMUSEUM_INGRESS ingress.us-west-2.staging.example.com
+chamber write kops CHARTMUSEUM_HOSTNAME charts.us-west-2.staging.example.com
+chamber exec kops -- helmfile -f /conf/kops/helmfile.yaml --selector namespace=kube-system,chart=chartmuseum sync
+chamber exec kops -- helmfile -f /conf/kops/helmfile.yaml --selector namespace=kube-system,chart=chartmuseum-api sync
+```
+{{% /dialog %}}
+
+These are some of the environment variables you may want to configure:
+
+* `CHARTMUSEUM_BASIC_AUTH_USER` - HTTP basic authenticate username
+* `CHARTMUSEUM_BASIC_AUTH_PASS` - HTTP basic authenticate password
+* `CHARTMUSEUM_API_BASIC_AUTH_USER` - HTTP basic authenticate username for `charts-api`
+* `CHARTMUSEUM_API_BASIC_AUTH_PASS` - HTTP basic authenticate password for `charts-api`
+* `CHARTMUSEUM_SECRET_NAME` - Secret name to store TLS generated with [Kube Lego]({{< relref "kubernetes-backing-services/tls-management/kube-lego-lets-encrypt.md" >}})
+* `CHARTMUSEUM_API_SECRET_NAME` - Secret name to store TLS generated with [Kube Lego]({{< relref "kubernetes-backing-services/tls-management/kube-lego-lets-encrypt.md" >}}) for `charts-api`
+* `CHARTMUSEUM_STORAGE_AMAZON_PREFIX` - Prefix path to store charts in S3 bucket
+
+Environment variables can be specified in the Geodesic Module's `Dockerfile` or using [Chamber]({{< relref "tools/chamber.md" >}}) storage, which is recommended for all secrets.
+
+### Install with Custom Helmfile
+
+Add to your [Kubernetes Backing Services](/kubernetes-backing-services) Helmfile this code
```

Comment:
Add this code to your [Kubernetes Backing Services](/kubernetes-backing-services) Helmfile: